### PR TITLE
Amend Just language entry

### PR DESCRIPTION
**Pull request by casey**
Sorry for not catching this in the initial review!

- Remove `Justfile` alias. `just` is the name of the tool and the language.
- Just looks for files called `justfile` in any case, so add the three most common cases: `JUSTFILE`, `Justfile`, and `justfile`.
- Add the `.just` extension, which is in common use. See this search: https://github.com/search?q=extension%3Ajust

### CODE REVIEW

**Review by mihaigalos — COMMENTED** [review]
@casey, what would you say if we refactor this to `Justfile` instead of `just` as file naming?
The way it currently shows up is this:
image

**Comment by mihaigalos** [comment]
Or capitalize the 1st letter?

**Comment by Alhadis** [comment]
> Or capitalize the 1st letter?

Yes, that's supposed to be the language's **human-readable name**, not its file extension or interpreter. Note that exceptions are made for non-standard casing if that's how the language's name is canonically spelt (e.g., `reStructuredText`, `mIRC Script`, `cURL Config`, et al).

> Remove `Justfile` alias. `just` is the name of the tool and the language.

Don't. This is a valid addition: it enables users to highlight Markdown code-blocks with `` ```justfile `` as well as `` ```just ``.

File: lib/linguist/languages.yml
Comment (Alhadis) [inline lib/linguist/languages.yml:7839]: > Add the `.just` extension, which is in common use. See this search: https://github.com/search?q=extension%3Ajust

Using GitHub's new Code Search engine, I only found 88 results for [`path:**/?*.just`](https://github.com/search?q=path%3A**%2F%3F*.just&type=code), which falls far short of [our real-world usage criteria](https://github.com/github/linguist/blob/59b2d88b2242e6062384e5fb876668cc30ead951/CONTRIBUTING.md#adding-an-extension-to-a-language) (at least 200 unique users/repos that use the new extension).

**Review by Alhadis — CHANGES_REQUESTED** [review]

**Comment by mihaigalos** [comment]
@casey: Can you please allow others' contributions to this PR? I wanted to implement some of the suggestions but couldn't since it's read-only.

File: lib/linguist/languages.yml
Comment (casey) [reply lib/linguist/languages.yml]: Done! I removed the `.just` extension.

Is there a way for me to access the new Code search engine? The link `https://github.com/search?q=path%3A**%2F%3F*.just&type=code` doesn't show any results.

**Comment by casey** [comment]
@mihaigalos I think of the name of the language as being `just`, as in `the just language`, and not `the justfile language`. Doing a quick search on google, "the make language" seems to be more common than "the makefile language", so this is consistent with make. Although, I note that `Makefile` is used by linguist. Overall I prefer `just`, and capitalizing it seems to be the convention with linguist, so I changed it to `Just`.

> Don't. This is a valid addition: it enables users to highlight Markdown code-blocks with ```justfile as well as ```just.

@Alhadis I'm not sure I see the value of adding an additional alias. Wouldn't it be better to have a single way of annotating markdown code blocks?

**Comment by Alhadis** [comment]
> Wouldn't it be better to have a single way of annotating markdown code blocks?

No, because this is the same lookup machinery responsible for interpreting `language:` operators in code-searches, or the value of `linguist-language=` in a `.gitattributes` file. I think we can all agree that `js` and `javascript` should both be mapped to the same language entry. Likewise for Just.

Remember, users who type `` ```justfile `` in the middle of a long-winded comment won't always catch themselves writing the "wrong" name. Some might even fail to notice that their annotated code-block is missing highlighting altogether. We should strive for intuitive, "DWIM" behaviour instead of being needlessly restrictive.

**Comment by Alhadis** [comment]
@casey FYI, I pushed a fix to your branch that updated the capitalisation of the `samples/just` directory—something that's notoriously fiddly to do on a case-insensitive filesystem (so, macOS and Windows). The `languages.yml` list is also meant to be ordered case-sensitively, so I fixed that as well.

**Review by Alhadis — APPROVED** [review]

**Comment by casey** [comment]
> @casey FYI, I pushed a fix to your branch that updated the capitalisation of the samples/just directory—something that's notoriously fiddly to do on a case-insensitive filesystem (so, macOS and Windows). The languages.yml list is also meant to be ordered case-sensitively, so I fixed that as well.

Thanks!

> Remember, users who type ```justfile in the middle of a long-winded comment won't always catch themselves writing the "wrong" name. Some might even fail to notice that their annotated code-block is missing highlighting altogether. We should strive for intuitive, "DWIM" behaviour instead of being needlessly restrictive.

Fair enough! I added the alias back.

**Comment by Alhadis** [comment]
Ace. Well, this looks good to me. Now we just need to wait for @lildude to give the GitHub Staffer's Seal-of-Approval™ so we can merge this. :shipit:

File: lib/linguist/languages.yml
Comment (lildude) [reply lib/linguist/languages.yml]: Yup: https://github.com/features/code-search/

**Review by lildude — APPROVED** [review]
LGTM. Thanks.

Note: this PR will not be merged until close to when the next release is made. See [here](https://github.com/github/linguist/blob/master/docs/troubleshooting.md#my-linguist-pr-has-been-merged-but-github-doesnt-reflect-my-changes) for more details.

**Comment by lildude** [comment]
Don’t worry about merging in master. I’ll do it again before merging.

File: lib/linguist/languages.yml
Comment (mihaigalos) [inline lib/linguist/languages.yml:3343]: @casey:  [This](https://github.com/casey/just#hiding-justfiles) talks about `.justfile` naming. Would it not make sense to also include it in this list?

Comment (nk9) [reply lib/linguist/languages.yml:3343]: Note that in the wild, users also frequently use `.just` as a file suffix for Justfiles.

Comment (lildude) [reply lib/linguist/languages.yml:3343]: Neither of these extensions are sufficiently popular on GitHub for inclusion right now. The `.just` extension was discussed earlier in this PR [here](https://github.com/github/linguist/pull/6167#pullrequestreview-1185649435).

Comment (mihaigalos) [reply lib/linguist/languages.yml:3343]: I understood the popularity requirement as something to be fulfilled for new languages.
The PR is not producing support for a new language (the language is already known to Linguist), but instead adding file aliases to an already known language.

Comment (lildude) [reply lib/linguist/languages.yml:3343]: The popularity requirement applies for all new extensions and filenames, regardless of whether it's a new or known language. See the details in CONTRIBUTING.md

**Pull request merged**
